### PR TITLE
[Merged by Bors] - feat: port RingTheory.Coprime.Lemmas

### PR DESCRIPTION
**Pull request by ericrbg**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Coprime/Lemmas.lean [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+variable {R : Type u} {I : Type v} [CommSemiring R] {x y z : R} {s : I → R} {t : Finset I}
+
+-- open BigOperators porting note: commented out locale
+
+section
+
+open Classical
+
+theorem Nat.is_coprime_iff_coprime {m n : ℕ} : IsCoprime (m : ℤ) n ↔ Nat.coprime m n :=
```
Comment (dupuisf) [inline Mathlib/RingTheory/Coprime/Lemmas.lean:36]:
```suggestion
theorem Nat.isCoprime_iff_coprime {m n : ℕ} : IsCoprime (m : ℤ) n ↔ Nat.coprime m n :=
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

File: Mathlib/RingTheory/Coprime/Lemmas.lean [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      Int.coe_nat_dvd.1 <| by
+        rw [Int.ofNat_one, ← H]
+        exact
+          dvd_add (dvd_mul_of_dvd_right (Int.coe_nat_dvd.2 <| Nat.gcd_dvd_left m n) _)
+            (dvd_mul_of_dvd_right (Int.coe_nat_dvd.2 <| Nat.gcd_dvd_right m n) _),
+    fun H ↦
+    ⟨Nat.gcdA m n, Nat.gcdB m n, by
+      rw [mul_comm _ (m : ℤ), mul_comm _ (n : ℤ), ← Nat.gcd_eq_gcd_ab, show _ = _ from H,
+        Int.ofNat_one]⟩⟩
+#align nat.is_coprime_iff_coprime Nat.is_coprime_iff_coprime
```
Comment (ericrbg) [inline Mathlib/RingTheory/Coprime/Lemmas.lean:48]:
```suggestion
#align nat.is_coprime_iff_coprime Nat.isCoprime_iff_coprime
```

**Pull request closed without merging**